### ISSUE TITLE
Approximation

### ISSUE THREAD

**Issue by rexpan** [4201715125/S4201715125V1#5]
- [x] General Approximation in commit 081588f292d154d302ec08747c4c6c785ade0f0e

**Comment by rexpan** [comment]
Bug:

``` csharp
Xs.Aggregate((lowX, Ix) => lowX.Union(Ix));
```

`Aggregate` causes error when `Xs` has no elements.

Changed to 

``` csharp
Xs.Aggregate(
    ((IEnumerable<double?[]>)new double?[][] { }),
    (lowX, Ix) => lowX.Union(Ix))
```

**Comment by rexpan** [comment]
Solved in commit cd26a31f5b7c0ec6b1bb37e81ffd55e66e113d18
